src/hid_win.c

```diff
@@ -42,18 +42,53 @@ is_fido(HANDLE dev)
 		goto fail;
 	}
 
-	if (caps.OutputReportByteLength != CTAP_MAX_REPORT_LEN + 1||
-	    caps.InputReportByteLength != CTAP_MAX_REPORT_LEN + 1) {
-		fido_log_debug("%s: unsupported report len", __func__);
+	if ((uint16_t)caps.UsagePage != 0xf1d0)
+		goto fail;
+
+	if (caps.OutputReportByteLength - 1 > CTAP_MAX_REPORT_LEN ||
+	    caps.OutputReportByteLength - 1 < CTAP_MIN_REPORT_LEN) {
+		fido_log_debug("%s: invalid output report len: %d", __func__,
+		    caps.OutputReportByteLength - 1);
+		goto fail;
+	}
+	if (caps.InputReportByteLength - 1 > CTAP_MAX_REPORT_LEN ||
+	    caps.InputReportByteLength - 1 < CTAP_MIN_REPORT_LEN) {
+		fido_log_debug("%s: invalid input report len: %d", __func__,
+		    caps.InputReportByteLength - 1);
 		goto fail;
 	}
```

Comment:
silly suggestion: could we turn the subtraction into an addition on the right side of the inequation, so we don't depend on the type of OutputReportByteLength/InputReportByteLength to handle the < 1 case?

Reply:
Should be done. I also added an explicit cast to `int` to the debug statements.